test/Dialect/FIRRTL/failConstProp.mlir

```diff
@@ -0,0 +1,90 @@
+// RUN: circt-opt -pass-pipeline='firrtl.circuit(firrtl-imconstprop)' --split-input-file  %s | FileCheck %s
+
+; XFAIL: *
+//propagate module chains not connected to the top
+firrtl.circuit "TopUnconnected"   {
+  firrtl.module @TopUnconnected(in %dummy: !firrtl.uint<1>) {
+  }
+  // CHECK-LABEL: firrtl.module @unconnectedChild1
+  firrtl.module @unconnectedChild1(out %out: !firrtl.uint<1>) {
+    %one_test = firrtl.instance @baz1  {name = "one"} : !firrtl.uint<1>
+    %zero_test = firrtl.instance @baz0  {name = "zero"} : !firrtl.uint<1>
+    %0 = firrtl.or %one_test, %zero_test : (!firrtl.uint<1>, !firrtl.uint<1>) -> !firrtl.uint<1>
+    // CHECK:  firrtl.connect %out, %c1_ui1_0 : !firrtl.uint<1>, !firrtl.uint<1>
```

Comment:
It would be good for tests like this to check the constant exists and pattern match it's name with the connect, so this isn't dependent on naming.